client/src/components/Navbar.js

```diff
@@ -1,16 +1,80 @@
-import React from 'react';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
 import AppBar from '@material-ui/core/AppBar';
 import Toolbar from '@material-ui/core/Toolbar';
 import Typography from '@material-ui/core/Typography';
+import { withStyles } from '@material-ui/core/styles';
+import Signup from './Signup';
+import AuthButtons from './AuthButtons';
+
+const styles = () => ({
+  root: {
+    flexGrow: 1,
+  },
+  grow: {
+    flexGrow: 1,
+  },
+});
+
+class Navbar extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      showLogin: false,
+      showSignup: false,
+    };
+  }
+
+  toggleModal = (modalName) => {
+    this.setState(prevState => ({
+      [modalName]: !prevState[modalName],
+    }));
+  };
+
+  render() {
+    const { classes, loggedInUser } = this.props;
+    const { showSignup } = this.state;
+
+    return (
+      <div className={classes.root}>
+        <AppBar position="static">
+          <Toolbar>
+            <Typography variant="title" color="inherit" className={classes.grow}>
+              Rattle Battle
+            </Typography>
+
+            {loggedInUser ? (
+              <Typography variant="subheading" color="inherit">
+                {`Hi, ${loggedInUser.username}`}
+              </Typography>
+            ) : (
+              <AuthButtons toggleModal={this.toggleModal} />
+            )}
+          </Toolbar>
+        </AppBar>
+        <Signup show={showSignup} toggleModal={this.toggleModal} />
```

Comment:
Here instead of signup signupDialog or signupModal seems more appropriate as I was going through the code it seems that signup is some sort of a div or form or button and my instant reaction was to why signup is shown for loggedInUser. But after I check the signup component then I got it that it is just a signupModal/ Dialog

Reply:
Acknowledged. I'll make the change.